First Assignment

khushalhiranii opened the pull request:
PR for first assignment:
1.calculator.c
2.crud_file.c

khushalhiranii committed:
assignment 1 pushed

kushrwt1 commented on crud_file.c:
```diff
@@ -0,0 +1,217 @@
+#include <stdio.h>
```
1. While running, Input process is not working properly(Asking for ID and name at the same time). Correct it

![image](https://github.com/user-attachments/assets/04319815-30f3-4417-980c-3f4c8f666993)

2.  User data is not displaying at all

![image](https://github.com/user-attachments/assets/25c3241d-dbee-482e-b7ec-cdf3954fcec7)

**It's not in the running state. Make it work to do all possible types of functions including Displaying users, deleting them, etc.**

3. Include error handling in your code.

kushrwt1 commented on crud_file.c:
```diff
@@ -0,0 +1,217 @@
+
+    while (fgets(buffer, sizeof(buffer), fp)) {
+        if (sscanf(buffer, "%[^,],%[^,],%d", user.id, user.name, &user.age) == 3) {
+            if (strcmp(user.id, id) == 0) {
```
Do not use inbuilt functions.
Make your own functions and use them and this comment is for all places where you have used inbuilt functions
Check and correct them

kushrwt1 commented on calculator.c:
```diff
@@ -0,0 +1,113 @@
+    return evaluateParsedExpression(numbers, operators, numCount, opCount);
+}
+
+int main() {
```
In main() function, there should be only function calling.
There should not be things like "taking inputs", etc.

kushrwt1 commented on crud_file.c:
```diff
@@ -0,0 +1,217 @@
+            case 2:
+                displayUsers();
+                break;
+            case 3: {
```
In switch statement cases, you should only call functions and not write business logic here or don't take taking inputs here

kushrwt1 commented on crud_file.c:
```diff
@@ -0,0 +1,217 @@
+
+// Add a new user to the database
+void addUser() {
+    FILE *fp = fopen("users.txt", "a");
```
Focus on variable naming which is understandable. Here, in place of "fp", a better name can be chosen.
This applies for all the variables in your code.

kushrwt1 commented on crud_file.c:
```diff
@@ -0,0 +1,217 @@
+
+// Update a user by ID
+void updateUser(const char *id) {
+    if (!deleteUser(id)) {
```
why we used deleteUser() function here.

kushrwt1 commented on calculator.c:
```diff
@@ -0,0 +1,113 @@
+    int i = 0, len = strlen(input), num = 0, isBuildingNumber = 0;
+
+    *numCount = 0;
+    *opCount = 0;
```
Focus on Variable naming throughout the code which is understandable.
Name in a way that its functionality can be understood just by reading name.

kushrwt1 commented on calculator.c:
```diff
@@ -0,0 +1,113 @@
+
+    printf("Enter a mathematical expression: ");
+    fgets(input, MAX_INPUT_SIZE, stdin);
+    input[strcspn(input, "\n")] = 0;
```
Don't use inbuilt Functions.
Built your own functions and use them.

kushrwt1 commented on calculator.c:
```diff
@@ -0,0 +1,113 @@
+    while (i < len) {
+        char ch = input[i];
+
+        if (isspace(ch)) {
```
Don't use inbuilt functions. Make your own logic.

kushrwt1 commented on calculator.c:
```diff
@@ -0,0 +1,113 @@
+    }
+}
+
+int evaluateParsedExpression(int numbers[], char operators[], int numCount, int opCount) {
```
Too much of logic in the single function. Make small functions and use them.

khushalhiranii committed:
Focused on :
- improved naming of functions and variables
- not using inbuilt functions
- only function calling in main()
- error handling in crud file
- updating the user data without deleting it before hand
- only function calls in switch statement cases

khushalhiranii committed:
removed inbuilt functions for input
